pkg/endpointmanager/manager.go

```diff
@@ -728,7 +728,24 @@ func (mgr *endpointManager) InitHostEndpointLabels(ctx context.Context) {
 		log.Error("Attempted to init host endpoint labels but host endpoint not set.")
 		return
 	}
-	ep.InitWithNodeLabels(ctx, launchTime)
+
+	mgr.initHostEndpointLabels(ctx, ep)
+}
+
+func (mgr *endpointManager) initHostEndpointLabels(ctx context.Context, ep *endpoint.Endpoint) {
+	// initHostEndpointLabels is executed by the daemon start hook, and
+	// at that point we are guaranteed that the local node has already
+	// been initialized, and this Get() operation returns immediately.
+	ln, err := mgr.localNodeStore.Get(ctx)
+	if err != nil {
```

Comment:
We could perhaps just no-op if context got cancelled here (e.g. fataling here might lead to unclear shutdown)

Reply:
Makes sense to me. I've changed it to just return immediately to avoid cluttering the shutdown process with possibly misleading error messages.